### PR TITLE
macro config parsing in core

### PR DESCRIPTION
**Pull request by C0W0**
# Description

Currently, there is no easy way for macro developers to render custom configs to the front end. This PR contains Typescript parsing logic, which runs independently of the Deno runtime to extract custom config templates defined by the macro developer into SettingManifests. We might need some corresponding frontend logic to test it out.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

*Note: make sure your files are formatted with rust-analyzer*

### CODE REVIEW

**Comment by Ynng** [comment]
I think this PR is directed at the wrong branch, `core_dev` and `core_main` are read only, we should merge to `dev` instead.

File: core/src/macro_executor.rs [hunk 1]
```diff
@@ -564,6 +570,326 @@ impl MacroExecutor {
...
+        let mut open_count = 0;
+        let mut close_count = 0;
+        let mut i = 0;
+        for &char_item in config_code.to_string().as_bytes().iter() {
+            if char_item == b'{' {
+                open_count += 1;
+            }
+            if char_item == b'}' {
+                close_count += 1;
+            }
```
Comment (CheatCod) [inline core/src/macro_executor.rs:605]: this can be replaced with an iterator

**Review by CheatCod — CHANGES_REQUESTED** [review]
More comments on how everything works will be great

File: core/src/macro_executor.rs (hunk 1 above)
Comment (C0W0) [reply core/src/macro_executor.rs:605]: Wait how...

**Review by CheatCod — CHANGES_REQUESTED** [review]
add test cases please

**Review by CheatCod — CHANGES_REQUESTED** [review]
looks like some files are changed not related to this PR? you should revert them if that's the case

**Comment by C0W0** [comment]
@CheatCod added comments and fixed those irrelevant files. Just needed to do a merge with dev to get up to date. Also added unit test

**Review by CheatCod — APPROVED** [review]

**Pull request merged**